include/groonga/groonga.h

```diff
@@ -436,14 +436,29 @@ typedef uint32_t grn_column_flags;
 #define GRN_OBJ_TABLE_DAT_KEY   (0x02)
 #define GRN_OBJ_TABLE_NO_KEY    (0x03)
 
-#define GRN_OBJ_KEY_MASK        (0x07 << 3)
-#define GRN_OBJ_KEY_UINT        (0x00 << 3)
-#define GRN_OBJ_KEY_INT         (0x01 << 3)
-#define GRN_OBJ_KEY_FLOAT       (0x02 << 3)
-#define GRN_OBJ_KEY_GEO_POINT   (0x03 << 3)
-
-#define GRN_OBJ_KEY_WITH_SIS    (0x01 << 6)
-#define GRN_OBJ_KEY_NORMALIZE   (0x01 << 7)
+/// Mask of `GRN_OBJ_KEY_*`
+#define GRN_OBJ_KEY_MASK (0x07 << 3)
+/// Unsigned integer
+/// (Used to create data types. Used to determine sorting method.)
+#define GRN_OBJ_KEY_UINT (0x00 << 3)
+/// Signed integer
+/// (Used to create data types. Used to determine sorting method.)
+#define GRN_OBJ_KEY_INT (0x01 << 3)
+/// Float
+/// (Used to create data types. Used to determine sorting method.)
+#define GRN_OBJ_KEY_FLOAT (0x02 << 3)
+/// Latitude and Longitude (\ref grn_geo_point)
+/// (Used to create data types. Used to determine sorting method.)
+#define GRN_OBJ_KEY_GEO_POINT (0x03 << 3)
+/// Semi Infinite String
+/// Used in backward matching search. (The tokens are huge.)
```

Comment:
```suggestion
/// Enable semi-infinite string support.
/// This is only available with \ref GRN_OBJ_TABLE_PAT_KEY.
/// You can use efficient suffix search with this but it requires more storage/memory
/// size. Because it generates many additional data for efficient suffix search implicitly.
```